### PR TITLE
feat: lint workflow templates

### PR DESCRIPTION
**Pull request by transt**
Add git workflow to argo lint understack workflow templates.

### CODE REVIEW

File: .github/workflows/lint-workflow-templates.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+          chmod +x argo-linux-amd64
+          sudo mv argo-linux-amd64 /usr/local/bin/argo
+          argo version
+
+      - name: Argo Lint Workflow Template
+        run: |
+          for file in workflows/argo-events/workflowtemplates/*.yaml; do
+              echo "Linting workflow template: $file"
+              argo lint $file
+          done
```
Comment (cardoe) [inline .github/workflows/lint-workflow-templates.yaml:42]:
```suggestion
        run: |
          ret=0
          for file in workflows/argo-events/workflowtemplates/*.yaml; do
              echo "Linting workflow template: $file"
              argo lint $file
              if [ $? -ne 0 ]; then
                  echo "::error file=$file,title=argo lint failed"
                  ret=1
              fi
          done
          exit $ret
```

Does `argo lint` return a non-zero exit code? Should we bubble up some more info?

see this doc for context: https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/workflow-commands-for-github-actions#setting-an-error-message

**Review by cardoe — COMMENTED** [review]
Just wondering how we'll make the workflow fail and report the error.

File: .github/workflows/lint-workflow-templates.yaml (hunk 1 above)
Comment (transt) [reply .github/workflows/lint-workflow-templates.yaml]: Updated

Comment (cardoe) [reply .github/workflows/lint-workflow-templates.yaml]: Nice!

**Review by cardoe — APPROVED** [review]

**Pull request merged**